Enhancement/6395 remove legacy report date params

eugene-manuilov opened the pull request:
## Summary

<!-- Please reference the issue this PR addresses in the following list. -->
Addresses issue:

- #6395 

## Relevant technical choices

<!-- Please describe your changes. -->

## PR Author Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 5.2 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [x] Ensure no unrelated changes are included.
- [x] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [x] Ensure the PR has the correct target branch.
- [x] Double-check that the PR is okay to be merged.
- [x] Ensure the corresponding issue has a ZenHub release assigned.
- [x] Add a changelog message to the issue.

techanvil reviewed (changes requested):
Hi @eugene-manuilov, nice work here.

There's one JSDoc comment that still references `dateRange` which should be updated, please see below.

Also, for completeness' sake, it's probably worth adding a `@since n.e.x.t Removed x parameter(s).` comment to the various modified functions/methods...

Here's that last `dateRange` reference:

https://github.com/google/site-kit-wp/blob/e5f6419deb311b31856e20324e5842abdb5a805a/assets/js/modules/analytics-4/datastore/report.js#L241-L242

eugene-manuilov commented:
> There's one JSDoc comment that still references `dateRange` which should be updated, please see below.

Good catch. Updated.

> Also, for completeness' sake, it's probably worth adding a `@since n.e.x.t Removed x parameter(s).` comment to the various modified functions/methods...

Technically saying we don't change the functionality of `getReport` selectors, we have just removed old arguments from javascript documentations. So, I think there is no need to add a new `@since n.e.x.t` section.

techanvil reviewed (approved):
Nice one @eugene-manuilov, LGTM!
